backend/app/modules/common/sablier_events_mapper.py

```diff
@@ -0,0 +1,169 @@
+        locks=list(
+            filter(
+                lambda lock: lock["__typename"] == EventType.LOCK,
+                lock_items_with_filters,
+            )
+        ),
+        unlocks=list(
+            filter(
+                lambda lock: lock["__typename"] == EventType.UNLOCK,
+                lock_items_with_filters,
+            )
+        ),
+    )
+
+
+def _apply_filters(
+    event_items: List[SablierEvent],
+    *,
+    from_timestamp: int,
+    to_timestamp: int,
+    inclusively: bool,
+) -> List[SablierEvent]:
+    copy_event_items = deepcopy(event_items)
+
+    if inclusively is True:
+        to_timestamp += 1
+
+    for item in event_items:
+        if from_timestamp and item["timestamp"] < from_timestamp:
+            copy_event_items.remove(item)
+        if to_timestamp and item["timestamp"] > to_timestamp:
+            copy_event_items.remove(item)
+
+    return copy_event_items
+
+
+def _process_create(
+    action: SablierAction, starting_deposit: int
+) -> Tuple[SablierEvent, int]:
+    amount = int(action["amountA"]) if action["amountA"] else 0
+    deposit_before = starting_deposit
+    starting_deposit += amount
+    lock_item = SablierEventLock(
+        __source=DepositSource.SABLIER,
+        __typename=EventType.LOCK.value,
+        amount=amount,
+        timestamp=int(action["timestamp"]),
+        transactionHash=action["hash"],
+        depositBefore=deposit_before,
+        user=action["addressB"],
+        type=SablierEventType.CREATE,
+    )
+    return lock_item, starting_deposit
+
+
+def _process_withdraw(
+    action: SablierAction, starting_deposit: int
+) -> Tuple[SablierEvent, int]:
+    amount = int(action["amountB"]) if action["amountB"] else 0
+    deposit_before = starting_deposit
+    starting_deposit -= amount
+    lock_item = SablierEventUnlock(
+        __source=DepositSource.SABLIER,
+        __typename=EventType.UNLOCK.value,
+        amount=amount,
+        timestamp=int(action["timestamp"]),
+        transactionHash=action["hash"],
+        depositBefore=deposit_before,
+        user=action["addressB"],
+        type=SablierEventType.WITHDRAW,
+    )
+    return lock_item, starting_deposit
+
+
+def _process_cancel(
+    action: SablierAction, starting_deposit: int
+) -> Tuple[SablierEvent, int]:
+    intact_amount = int(action["amountB"]) if action["amountB"] else 0
+    cancelled_amount = int(action["amountA"]) - intact_amount
+    deposit_before = starting_deposit
+    starting_deposit = intact_amount
+    lock_item = SablierEventUnlock(
+        __source=DepositSource.SABLIER,
+        __typename=EventType.UNLOCK.value,
+        amount=cancelled_amount,
+        timestamp=int(action["timestamp"]),
+        transactionHash=action["hash"],
+        depositBefore=deposit_before,
+        user=action["addressB"],
+        type=SablierEventType.CANCEL,
+    )
+    return lock_item, starting_deposit
+
+
+def _convert(actions: List[SablierAction]) -> List[SablierEvent]:
```

Comment:
fyi: sablier doesn't publish any kind of `depositBefore` field, thus we have to compute it dynamically; intactAmount doesn't work here since it has a static value computed once at the end of stream and is the same for all the events